docs/operating/concepts/podopslifecycle.md

```diff
@@ -0,0 +1,51 @@
+
+#### Story 2
+
+As a developer that maintain a system that provide pod operations like replace, delete, scale out, and so on, I should add the label `operating.podopslifecycle.kusionstack.io/<id>=<time>` and `operation-type.podopslifecycle.kusionstack.io/<id>=<type>` at the same time when I want to operate a pod, and I should remove the label `operating.podopslifecycle.kusionstack.io/<id>=<time>` and `operation-type.podopslifecycle.kusionstack.io/<id>=<type>` at the same time when the operation is completed. If I want to cancel the operation, I should add the label `undo-operation-type.podopslifecycle.kusionstack.io/<id>=<type>`.
+
+The sequence diagram below describe how to offline a pod.
+
+![](https://mermaid.ink/svg/pako:eNqtUk1LAzEU_CuPnNvtfakLQo-KojfJ5TV524bmy-RtZSn97yZbqSJWPTSnZBhmJsMchAqaRCsyvQ7kFa0MbhI66SWj4pBgRXuyIVKqUMTERpmInuExaOmhnDMD5l1X4RbuTGZAr-ENWW0v0m61BotrslBxZOM3TQw6xGxNT2pUlprdkE3wuaTZNSYslkZ3N0s2jroZLNdp0VUb_C4U_JzHSP9UK8wOkIG3BBkdQdVvTrFL0CnwOX753RenvyxqNLpc1EPfW-Npco6_FPpELuzpOmWlH7Su0ZeYCUfJodFlT4f6FSkKw5EUbblq6nGwLIX0x0LFgcPz6JVoOQ00E0PUpc2P-Ym2R5sLWpb2EsLnm7Qpo7w_bXaa7vEdMZH5Yg)
+
+#### Story 3
+
+As a developer that focus on pod operation observability, I can use the `<id>=<time>` and `<id>=<type>` in the labels to tracing a pod. The `<time>` is a unix nano time, and the `<type>` is a string that describe the operation type, and the `<id>` is a string that used in the whole operation lifecycle.
+
+## Design Details
+
+1. Podopslifecycle mechanism is provided by a mutating webhook server and a controller. The mutating webhook server will chage the labels at the right time, and the controller will set the readinessgate `pod.kusionstack.io/service-ready` to true or false if necessary. The controller will also chage the label at some time.
+2. The label `operating.podopslifecycle.kusionstack.io/<id>=<time>` and `operation-type.podopslifecycle.kusionstack.io/<id>=<type>` will be validated by a validating webhook server, they must be added or removed at the same time by the operation controller.
+3. Traffic controller should turn the traffic on or off based on label `prepare.podopslifecycle.kusionstack.io/<id>=<time>` and `complete.podopslifecycle.kusionstack.io/<id>=<time>`.
+4. Protection finalizer names must have prefix `prot.podopslifecycle.kusionstack.io`. They are used to determine whether the traffic has been completely removed or is fully prepared.
+5. The special label `podopslifecycle.kusionstack.io/service-available` indicate a pod is available to serve.
+6. We can use the message `<id>=<time>` and `<id>=<type>` in the labels to tracing a pod. The `<time>` is a unix time.
+
+Below we use a sequence diagram to show how to use podopslifecycle mechanism to avoid traffic loss. You can also use this podopslifecycle mechanism to do others things, for example, to prevent tasks to be interrupted when they are need to run for a long time.
```

Comment:
I think we can use the pkg (https://github.com/KusionStack/operating/blob/main/pkg/controllers/utils/podopslifecycle/utils.go) to show a simple demo that how a controller interact with podOpsLifecycle.

Reply:
OK, I will add it to manual.